Reproduce convergence edit_config_for_scaling_group intermittent failure

On [CI server running convergence tests](https://as-jenkins.rax.io/job/_ci-integration-convergence-gating/39/console), transaction ID bd4358cf-91eb-4d5e-9432-e39994f99849:

```
Request failed: Unhandled Error: AttributeError("'NoneType' object has no attribute 'tenant_id'",)", "otter_facility": "otter.rest.configs.edit_config_for_scaling_group", "request_status": "failed", "scaling_group_id": "8c5bbb20-9834-4fae-b3b9-92b396da9852", "tenant_id": "000000", "traceback": "Traceback (most recent call last):
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 578, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/models/cass.py", line 640, in wrapper
    d.addCallback(lambda ts: func(ts, *args))
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 307, in addCallback
    callbackKeywords=kw)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 296, in addCallbacks
    self._runCallbacks()
--- <exception caught here> ---
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 578, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/models/cass.py", line 640, in <lambda>
    d.addCallback(lambda ts: func(ts, *args))
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/models/cass.py", line 761, in _write_state
    assert (new_state.tenant_id == self.tenant_id and
exceptions.AttributeError: 'NoneType' object has no attribute 'tenant_id'
```

and 

```
Error when performing convergence: TypeError("Attribute 'server_id' must be an instance of 'str'.",)", "otter_facility": "otter.rest.configs.edit_config_for_scaling_group", "otter_msg_type": "convergence-perform-error", "scaling_group_id": "8c5bbb20-9834-4fae-b3b9-92b396da9852", "tenant_id": "000000", "traceback": "Traceback (most recent call last):
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/effect/_continuation.py", line 25, in bounce
    trampoline(func, *args, **kwargs)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/effect/_continuation.py", line 56, in trampoline
    f(bouncer, *args, **kwargs)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/effect/_base.py", line 134, in _run_callbacks
    bouncer.bounce(_run_callbacks, chain, result)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/effect/_base.py", line 83, in guard
    return (True, sys.exc_info())
--- <exception caught here> ---
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/effect/_base.py", line 81, in guard
    return (False, f(*args, **kwargs))
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/convergence/service.py", line 82, in got_all_data
    steps = plan(desired_group_state, servers, lb_nodes, now)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/convergence/planning.py", line 353, in plan
    steps = converge(desired_group_state, servers, lb_nodes, now)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/convergence/planning.py", line 219, in converge
    servers_to_delete + draining_servers))
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/convergence/planning.py", line 216, in drain_and_delete_a_server
    now)
  File "/opt/otter/.ve/local/lib/python2.7/site-packages/otter/convergence/planning.py", line 144, in _drain_and_delete
    return lb_draining_steps + [DeleteServer(server_id=server.id)]
  File "<characteristic generated init 9be01cf021e1054cef16f7e497966332e68a1a2e>", line 14, in characteristic_init
    raise TypeError("Attribute 'server_id' must be an instance of 'str'.")
exceptions.TypeError: Attribute 'server_id' must be an instance of 'str'.
```

And on that group, there were no servers.  :|  This seems intermittent, because running the CI job again did not result in this error.

```
======================================================================
FAIL: test_get_group_after_deletion (test_repo.autoscale.functional.scaling_group.test_scaling_group_negative.ScalingGroupNegative)
Negative Test: Get group when group is deleted should fail with 404
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/ci-integration-tests/.cf-ve/local/lib/python2.7/site-packages/test_repo/autoscale/functional/scaling_group/test_scaling_group_negative.py", line 216, in test_get_group_after_deletion
    ' for group {0}'.format(group.id))
AssertionError: 500 != 204 : Delete group failed for group 80d75dd5-e4a4-4817-a657-62bc33d96cf8
```

--- a/otter/convergence/model.py
+++ b/otter/convergence/model.py
@@ -377,7 +377,7 @@ class IDrainable(Interface):
 
 
 @implementer(ILBDescription)
-@attributes([Attribute("lb_id", instance_of=str),
+@attributes([Attribute("lb_id", instance_of=basestring),
              Attribute("port", instance_of=int),
              Attribute("weight", default_value=1, instance_of=int),
              Attribute("condition", default_value=CLBNodeCondition.ENABLED,
@@ -416,9 +416,9 @@ class CLBDescription(object):
 
 
 @implementer(ILBNode, IDrainable)
-@attributes([Attribute("node_id", instance_of=str),
+@attributes([Attribute("node_id", instance_of=basestring),
              Attribute("description", instance_of=CLBDescription),
-             Attribute("address", instance_of=str),
+             Attribute("address", instance_of=basestring),
              Attribute("drained_at", default_value=0.0, instance_of=float),
              Attribute("connections", default_value=None)])
 class CLBNode(object):
@@ -467,7 +467,7 @@ class CLBNode(object):
 
 
 @implementer(ILBDescription)
-@attributes([Attribute("lb_id", instance_of=str)])
+@attributes([Attribute("lb_id", instance_of=basestring)])
 class RCv3Description(object):
     """
     Information representing a RackConnect V3/server mapping: how a particular
@@ -489,9 +489,9 @@ class RCv3Description(object):
 
 
 @implementer(ILBNode)
-@attributes([Attribute("node_id", instance_of=str),
+@attributes([Attribute("node_id", instance_of=basestring),
              Attribute("description", instance_of=RCv3Description),
-             Attribute("cloud_server_id", instance_of=str)])
+             Attribute("cloud_server_id", instance_of=basestring)])
 class RCv3Node(object):
     """
     A RackConnect V3 node.

--- a/otter/convergence/steps.py
+++ b/otter/convergence/steps.py
@@ -156,7 +156,7 @@ class CreateServer(object):
 
 
 @implementer(IStep)
-@attributes([Attribute('server_id', instance_of=str)])
+@attributes([Attribute('server_id', instance_of=basestring)])
 class DeleteServer(object):
     """
     A server must be deleted.
@@ -182,9 +182,9 @@ class DeleteServer(object):
 
 
 @implementer(IStep)
-@attributes([Attribute('server_id', instance_of=str),
-             Attribute('key', instance_of=str),
-             Attribute('value', instance_of=str)])
+@attributes([Attribute('server_id', instance_of=basestring),
+             Attribute('key', instance_of=basestring),
+             Attribute('value', instance_of=basestring)])
 class SetMetadataItemOnServer(object):
     """
     A metadata key/value item must be set on a server.
@@ -241,7 +241,7 @@ def _check_clb_422(*regex_matches):
 
 
 @implementer(IStep)
-@attributes([Attribute('lb_id', instance_of=str),
+@attributes([Attribute('lb_id', instance_of=basestring),
              Attribute('address_configs', instance_of=PSet)])
 class AddNodesToCLB(object):
     """
@@ -284,7 +284,7 @@ class AddNodesToCLB(object):
 
 
 @implementer(IStep)
-@attributes([Attribute('lb_id', instance_of=str),
+@attributes([Attribute('lb_id', instance_of=basestring),
              Attribute('node_ids', instance_of=PSet)])
 class RemoveNodesFromCLB(object):
     """
@@ -351,8 +351,8 @@ def _clb_check_bulk_delete(lb_id, attempted_nodes, result):
 
 
 @implementer(IStep)
-@attributes([Attribute('lb_id', instance_of=str),
-             Attribute('node_id', instance_of=str),
+@attributes([Attribute('lb_id', instance_of=basestring),
+             Attribute('node_id', instance_of=basestring),
              Attribute('condition', instance_of=NamedConstant),
              Attribute('weight', instance_of=int),
              Attribute('type', instance_of=NamedConstant)])

